### ISSUE TITLE
[BUG] progress bar: bogus estimations

### ISSUE THREAD

**Issue by nschloe** [willmcgugan/rich#378]
The computation of the remaining time often produces bogus results for me. Here, when nesting two progress bars, I'm even getting _negative_ times. MWE:
```python
from rich.progress import Progress
import time

m = 10
n = 5

with Progress() as progress:
    task1 = progress.add_task("Overall", total=m)
    task2 = progress.add_task("Kernels", total=n)

    for _ in range(m):
        progress.update(task2, completed=0)
        for j in range(n):
            time.sleep(2 ** j)
            progress.update(task2, advance=1)
        progress.update(task1, advance=1)
```

![screenshot](https://user-images.githubusercontent.com/181628/96718581-8020d180-13a8-11eb-9ccf-d2b4b5037a93.png)

**Comment by willmcgugan** [comment]
Are you adjusting the total after the task is started?

I'll need to see code to diagnose this.

**Comment by nschloe** [comment]
@willmcgugan Added MWE to the original post.

**Comment by willmcgugan** [comment]
Can you try with the latest version please.

**Comment by nschloe** [comment]
This is with
```
python3 -c 'from importlib import metadata; print(metadata.version("rich"))'
```
```
9.0.1
```

**Comment by willmcgugan** [comment]
It's because completed is going backwards. Replace `progress.update(task2, completed=0)` with `progress.reset(task2)`.

I'll look in to doing an automatic reset when the value completed goes backwards.

**Comment by nschloe** [comment]
That fixes it, thanks. Should I leave this open?

**Comment by willmcgugan** [comment]
Great. Yes, leave it open for now.

**Comment by willmcgugan** [comment]
Fixed in 9.1.0. Progress.reset is still the recommended way though.
